db/migrate/20210614145351_create_friendly_id_slugs.rb

```diff
@@ -0,0 +1,32 @@
+# frozen_string_literal: true
+
+MIGRATION_CLASS =
+  if ActiveRecord::VERSION::MAJOR >= 5
+    ActiveRecord::Migration["#{ActiveRecord::VERSION::MAJOR}.#{ActiveRecord::VERSION::MINOR}"]
+  else
+    ActiveRecord::Migration
+  end
+
+class CreateFriendlyIdSlugs < MIGRATION_CLASS
+  def change
+    create_table :friendly_id_slugs do |t|
+      t.string   :slug,           null: false
+      t.integer  :sluggable_id,   null: false
+      t.string   :sluggable_type, limit: 50
+      t.string   :scope
+      t.datetime :created_at
+    end
+    add_index :friendly_id_slugs, %i[sluggable_type sluggable_id]
+    add_index :friendly_id_slugs, %i[slug sluggable_type], length: { slug: 140, sluggable_type: 50 }
+    add_index :friendly_id_slugs, %i[slug sluggable_type scope], length: { slug: 70, sluggable_type: 50, scope: 70 }, unique: true
+
+    add_column :schools, :slug, :string
+    add_index :schools, :slug, unique: true
+
+    # This takes > 4 minutes to run which fails deploys - run manually on
+    # deployed environments
+    if Rails.env.development? || Rails.env.test?
+      School.find_each(&:save)
+    end
```

Comment:
uhhhhhhh, I would really prefer not to have to run it manually. Can we bump the time limit? What is failing exactly? Our github action? paas?

Reply:
dunno, this was decided after conversations with @ethanmills and @jamesrose. I was way out of my depth in them, so this is probably a question that's best for them

Reply:
What's wrong with running this manually? 60s is the limit on a deploy going green - has to 200 from /check within this time

Reply:
Anything that needs to be ran manually is not ran at some point.

Although I guess for all new environments, the post create hook (which this is triggering) should run as part of the school import job, so we don't need this in the migration. Is that correct?

If so, it sounds like bumping the time for paas is not worth the effort, so let's keep it as is and make sure we remember to run slugs on dev, staging, sandbox and prod.

Reply:
It _only_ needs running on production, staging, and probably sandbox. The develop branches are reset not migrated, and this will run on people's local machines.

Reply:
It will also cause the site to fail pretty catastrophically if it isn't ran, so it's not like it will be causing subtle issues for a while that won't get noticed